compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -418,9 +420,40 @@ public void visit(BLangRecordTypeNode recordTypeNode) {
             rewrite(field, recordTypeNode.typeDefEnv);
         }
         recordTypeNode.restFieldType = rewrite(recordTypeNode.restFieldType, env);
+        generateClosuresForDefaultValuesInTypeRefs(recordTypeNode, typeSymbol);
         result = recordTypeNode;
     }
 
+    private void generateClosuresForDefaultValuesInTypeRefs(BLangRecordTypeNode recordTypeNode,
+                                                            BTypeSymbol typeSymbol) {
+        Map<String, BInvokableSymbol> defaultValues = ((BRecordTypeSymbol) typeSymbol).defaultValues;
+        String typeName = recordTypeNode.symbol.name.value;
+        for (BLangType type : recordTypeNode.typeRefs) {
```

Comment:
We seem to be creating quite a few closures unnecessarily. 

1. Isn't the problem only with inclusions from a different module? Why are we creating closures for defaults from all inclusions?

For the following points, inclusions can be from different modules once 1. is addressed.

2. For

```ballerina
type Foo record {|
    int a = 1;
    int b = 2;
    int c = 3;
|};

type Bar record {|
    *Foo;
    int b = 22;
    int c;
|};
```

We are creating a closure wrapping c's default value closure unnecessarily. That default value closure (Foo's c field's) is not required here.

3. For

```ballerina
type Foo record {|
    int a = 1;
|};

type Bar record {|
    *Foo;
|};

type Baz record {|
    *Foo;
|};
```

closures wrapping a's closure are created twice. Can't we avoid (or at least minimize) this using some combination of the included record's fully-qualified identifier and field name?

Reply:
I would fix at least 2. Easy fix, right? Just need to check by explicitly-specified fields rather than those with defaults.

Reply:
Yes. Will update that.

Reply:
Udapted https://github.com/ballerina-platform/ballerina-lang/pull/42428/commits/52836fd3b4c4d080aec543190c4d8af0b9c80665